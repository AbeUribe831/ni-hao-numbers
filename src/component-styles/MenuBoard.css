.flex-column {
    display: flex;
    flex-direction: column;
}
.flex-row {
    display: flex;
    flex-direction: row;
    align-self: center;
}
.welcome-text {
    color: white;
    text-align: center;
}
.no-padding-margin {
    padding: 0;
    margin: 0;
    width: 5em;
}
.underline-button {
    font-family: 'sans-serif';
    color: white;
    text-align: center;
    background: none;
    border: none;
    cursor:pointer;
    margin: 0px 0px 7px 0px;
    padding: 0;
    text-decoration: underline;
    font-size: medium;
}
.margin-top {
    margin-top: 0.5em;
}
.sub-titles {
    margin: 0;
    color: white;
    font-weight: bold;
}
.underline-button:hover {
    color: white;
    background-color: #07CDB6;
    border-radius: 2px;
}
.question-button {
    border: none;
    height: 2.5em;
    width: 8.5em;
    border-radius: 35%;
    font-weight: bold;
}
p {
    font-family: sans-serif;
}
input {
    font-family: sans-serif;
    padding: 0%;
    border: none;
    border-bottom: 1px solid;
    width: 7em;
    height: 2em;
    color: white;
    background: none;
}
input:focus {
    outline: none;
    border-bottom-color: #cd071e;
}
select {
    font-family: sans-serif;
    height: 2em;
    background: none;
    border: none;
    color: white;
    border-bottom: 1px solid;
}
select:focus {
    outline: none;
    border-bottom-color: #cd071e;
}