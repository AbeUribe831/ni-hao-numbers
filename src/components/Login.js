import '../App.css'
export default function Login() {
    return (
        <div className='main-board'>
        <h3 style={{textAlign : 'center'}}>login</h3>
        </div>

    )
}