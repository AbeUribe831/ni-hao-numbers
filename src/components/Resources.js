import React from 'react'
import '../component-styles/MainBoard.css'
export default function Resources() {
    return (
        <div className='main-board' style={{flex: '2'}}>
        <h3 style={{textAlign : 'center'}}>Resources</h3>
        </div>

    )
}