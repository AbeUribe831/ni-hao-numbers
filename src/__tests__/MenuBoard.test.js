/**
 * @jest-environment jsdom
 */
import React from 'react';
// import renderer from 'react-test-renderer';
import { MenuBoard, NumberOptions } from '../components/MenuBoard';
import { MoreOptions } from '../components/MenuBoard';
import { Options } from '../components/MenuBoard';
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json'

Enzyme.configure({adapter: new Adapter()});

let number_options_props;
let more_options_props;
let options_props;
let menu_board_props;
const mock_function = jest.fn();

beforeEach(() => {
    number_options_props = {
        isMobile:false,
        minBound:'0',
        updateMinBound:jest.fn(),
        updateMinBoundBlur:jest.fn(),
        confirmMinIsNumber:jest.fn(),
        maxBound:'10',
        decimalPlacement:'0',
        chnCharType:'sc',
        updateMaxBound:jest.fn(),
        updateMaxBoundBlur:jest.fn(),
        updateDecimalPlacement:jest.fn(),
        confirmMaxIsNumber:jest.fn(),
        howMany:'5',
        updateHowMany:jest.fn(),
        updateHowManyBlur:jest.fn(),
        updateChnCharType:jest.fn(),
        confirmHowMany:jest.fn(),
    };
    more_options_props = {
        isMobile:false,
        questions:{
            listen: false,
            readNumber: true,
            readCharacter: true
        },
        started:false,
        onClickMoreOptions:jest.fn(), 
        updateQuestions:jest.fn(),
        updateAnswers:jest.fn(),
        resetQAndA:jest.fn()
    };
        options_props = {
            isMobile:false,
            minBound:'0',
            maxBound:'10',
            howMany:'6',
            questions:{
                listen: false,
                readNumber: true,
                readCharacter: true
            },  
            answers:{speak: false},
            started:false,
            decimalPlacement:'0',
            chnCharType:'sc',
            onClickMoreOptions:jest.fn(), 
            moreOptions:false, 
            updateDecimalPlacement:jest.fn(),
            updateQuestions:jest.fn(),
            updateAnswers:jest.fn(),
            updateMinBound:jest.fn(),
            updateMaxBound:jest.fn(),
            updateHowMany:jest.fn(),
            updateChnCharType:jest.fn(),
            confirmMinIsNumber:jest.fn(),
            confirmMaxIsNumber:jest.fn(),
            confirmHowMany:jest.fn(),
            resetQAndA:jest.fn(),
            onClickStart:jest.fn(),
        };
        menu_board_props = {
            isMobile:false,
            minBound:'0',
            maxBound:'10',
            howMany:'5',
            questions:{
                listen: false,
                readNumber: true,
                readCharacter: true
            },
            answers:{speak: false},
            started:false,
            decimalPlacement:'0',
            chnCharType:'sc',
            onClickMoreOptions:jest.fn(),
            moreOptions:false,
            updateDecimalPlacement:jest.fn(),
            updateQuestions:jest.fn(),
            updateAnswers:jest.fn(),
            updateMinBound:jest.fn(),
            updateMinBoundBlur:jest.fn(),
            updateMaxBound:jest.fn(),
            updateMaxBoundBlur:jest.fn(),
            updateHowMany:jest.fn(),
            updateHowManyBlur:jest.fn(),
            updateChnCharType:jest.fn(),
            confirmMinIsNumber:jest.fn(),
            confirmMaxIsNumber:jest.fn(),
            confirmHowMany:jest.fn(),
            resetQAndA:jest.fn(),
            onClickStart:jest.fn(),
        }
});
afterEach(() => {
    jest.clearAllMocks();
})
describe('rendering components', () => {
    test('NumberOptions renders when isMobile is false', () => {
        Enzyme.shallow(
            <NumberOptions
                {...number_options_props}
            />
        );
    });
    test('NumberOptions renders when isMobile is true', () => {
        number_options_props.isMobile = true;
        Enzyme.shallow(
            <NumberOptions
                {...number_options_props}
            />
        );
    });
    test('MoreOptions renders when isMobile is false', () => {
        Enzyme.shallow(
            <MoreOptions
            {...more_options_props}
            />
        );
    });
    test('MoreOptions renders when isMobile is true', () => {
        more_options_props.isMobile = true;
        Enzyme.shallow(
            <MoreOptions
            {...more_options_props}
            />
        );
    });
    test('Options renders when isMobile is false', () => {
        Enzyme.shallow(
            <Options
                {...options_props}
            />
        );
    });
    test('MenuBoard renders when isMobile is false', () => {
        Enzyme.shallow(
            <MenuBoard
                {...menu_board_props}
            />
        );
    });
    test('MenuBoard renders when isMobile is true', () => {
        menu_board_props.isMobile=true
        Enzyme.shallow(
            <MenuBoard
                {...menu_board_props}
            />
        );
    });
})
describe('passing props', () => {
    test('NumberOptions passes props', () => {
        const wrapper = Enzyme.mount(
            <NumberOptions
                {...number_options_props}
            />
        );
        expect(wrapper.props().isMobile).toBe(false);
        expect(wrapper.props().minBound).toBe('0');
        expect(typeof wrapper.props().updateMinBound === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().confirmMinIsNumber === typeof mock_function).toBe(true);
        expect(wrapper.props().maxBound).toBe('10');
        expect(wrapper.props().decimalPlacement).toBe('0');
        expect(wrapper.props().chnCharType).toBe('sc');
        expect(typeof wrapper.props().updateMaxBound === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().updateDecimalPlacement === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().confirmMaxIsNumber === typeof mock_function).toBe(true);
        expect(wrapper.props().howMany).toBe('5');
        expect(typeof wrapper.props().updateHowMany === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().updateChnCharType === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().confirmHowMany === typeof mock_function).toBe(true);
    });
    test('MoreOptions passes props', () => {
        const wrapper = Enzyme.mount(
            <MoreOptions
                {...more_options_props}
            />
        );
        expect(wrapper.props().isMobile).toBe(false);
        expect(wrapper.props().questions).toStrictEqual({listen: false, readNumber: true, readCharacter: true})
        expect(wrapper.props().started).toBe(false);
        expect(typeof wrapper.props().onClickMoreOptions === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().updateQuestions === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().updateAnswers === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().resetQAndA === typeof mock_function).toBe(true);
    });
    test('Options passes props', () => {
        const wrapper = Enzyme.mount(
            <Options
                {...options_props}
            />
        );
        expect(wrapper.props().isMobile).toBe(false);
        expect(wrapper.props().questions).toStrictEqual({listen: false, readNumber: true, readCharacter: true})
        expect(wrapper.props().answers.speak).toBe(false);
        expect(wrapper.props().started).toBe(false);
        expect(wrapper.props().decimalPlacement).toBe('0');
        expect(wrapper.props().chnCharType).toBe('sc');
        expect(typeof wrapper.props().onClickMoreOptions === typeof mock_function).toBe(true);
        expect(wrapper.props().moreOptions).toBe(false);
        expect(typeof wrapper.props().updateDecimalPlacement === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().updateQuestions === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().updateAnswers === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().updateMinBound === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().updateMaxBound === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().confirmMinIsNumber === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().confirmMaxIsNumber === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().confirmHowMany === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().resetQAndA === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().onClickStart === typeof mock_function).toBe(true);
    });
    test('MenuBoard passes props', () => {
        const wrapper = Enzyme.mount(
            <MenuBoard
                {...menu_board_props}
            />
        );
        expect(wrapper.props().isMobile).toBe(false);
        expect(wrapper.props().minBound).toBe('0');
        expect(wrapper.props().maxBound).toBe('10');
        expect(wrapper.props().howMany).toBe('5');
        expect(wrapper.props().questions).toStrictEqual({listen: false, readNumber: true, readCharacter: true})
        expect(wrapper.props().answers.speak).toBe(false);
        expect(wrapper.props().started).toBe(false);
        expect(wrapper.props().decimalPlacement).toBe('0');
        expect(wrapper.props().chnCharType).toBe('sc');
        expect(typeof wrapper.props().onClickMoreOptions === typeof mock_function).toBe(true);
        expect(wrapper.props().moreOptions).toBe(false);
        expect(typeof wrapper.props().updateDecimalPlacement === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().updateQuestions === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().updateAnswers === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().updateMinBound === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().updateMinBoundBlur === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().updateMaxBound === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().updateMaxBoundBlur === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().updateHowMany === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().updateHowManyBlur === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().updateChnCharType === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().confirmMinIsNumber === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().confirmMaxIsNumber === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().confirmHowMany === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().resetQAndA === typeof mock_function).toBe(true);
        expect(typeof wrapper.props().onClickStart === typeof mock_function).toBe(true);
    })
});
// Options will have no test in logic, clicking moreOptions method is in MenuBoard and passed to MoreOptions
describe('logic', () => {
    const red_color = '#cd071e';

    test('MoreOptions colors change based on props and clicks are noted', () => {
        const wrapper = Enzyme.mount(
            <MoreOptions
                {...more_options_props}
            />
        );
        // check color and backgroundColor of button before and after clicked
        let read_character_button = wrapper.find('#read-character');
        let read_number_button = wrapper.find('#read-number');
        let listen_button = wrapper.find('#listen');
        read_character_button.simulate('click');
        // readCharacter true 
        expect(read_character_button.props().style.color).toBe('white');
        expect(read_character_button.props().style.backgroundColor).toBe(red_color);
        expect(more_options_props.updateQuestions).toHaveBeenCalledTimes(1);
        // readNumber true
        read_number_button.simulate('click');
        expect(read_number_button.props().style.color).toBe('white');
        expect(read_number_button.props().style.backgroundColor).toBe(red_color);
        expect(more_options_props.updateQuestions).toHaveBeenCalledTimes(2);
        // listen false 
        listen_button.simulate('click');
        expect(listen_button.props().style.color).toBe(red_color);
        expect(listen_button.props().style.backgroundColor).toBe('white');
        expect(more_options_props.updateQuestions).toHaveBeenCalledTimes(3);

        // reverse the colors in MoreOptions
        more_options_props.questions.readCharacter = false;
        more_options_props.questions.readNumber = false;
        more_options_props.questions.listen = true;
        const rev_color_wrapper = Enzyme.mount(
            <MoreOptions
                {...more_options_props}
            />
        );
        // check color and backgroundColor of button before and after clicked
        read_character_button = rev_color_wrapper.find('#read-character');
        read_number_button = rev_color_wrapper.find('#read-number');
        listen_button = rev_color_wrapper.find('#listen');
        read_character_button.simulate('click');
        // readCharacter false
        expect(read_character_button.props().style.color).toBe(red_color);
        expect(read_character_button.props().style.backgroundColor).toBe('white');
        expect(more_options_props.updateQuestions).toHaveBeenCalledTimes(4);
        // readNumber false 
        read_number_button.simulate('click');
        expect(read_number_button.props().style.color).toBe(red_color);
        expect(read_number_button.props().style.backgroundColor).toBe('white');
        expect(more_options_props.updateQuestions).toHaveBeenCalledTimes(5);
        // listen true
        listen_button.simulate('click');
        expect(listen_button.props().style.color).toBe('white');
        expect(listen_button.props().style.backgroundColor).toBe(red_color);
        expect(more_options_props.updateQuestions).toHaveBeenCalledTimes(6);

        const hide_button = wrapper.find('#hide-button');
        hide_button.simulate('click');
        expect(more_options_props.onClickMoreOptions).toHaveBeenCalledTimes(1);
    });

    test('NumberOptions input changes are noted', () => {
        const wrapper = Enzyme.mount(
            <NumberOptions
                {...number_options_props}
            />
        );
        const min_bound_input = wrapper.find({name: 'minBound'});
        const max_bound_input = wrapper.find({name: 'maxBound'});
        const how_many_input = wrapper.find({name: 'howMany'});

        min_bound_input.simulate('change', {target: {value: '3'}});
        max_bound_input.simulate('change', {target: {value: '14'}});
        how_many_input.simulate('change', {target: {value: '7'}});
        expect(wrapper.props().updateMinBound).toHaveBeenCalledTimes(1);
        expect(wrapper.props().updateMaxBound).toHaveBeenCalledTimes(1);
        expect(wrapper.props().updateHowMany).toHaveBeenCalledTimes(1);

        // press enter so this works
        min_bound_input.simulate('keyDown');
        max_bound_input.simulate('keyDown');
        how_many_input.simulate('keyDown');
        expect(wrapper.props().confirmMinIsNumber).toHaveBeenCalledTimes(1);
        expect(wrapper.props().confirmMaxIsNumber).toHaveBeenCalledTimes(1);
        expect(wrapper.props().confirmHowMany).toHaveBeenCalledTimes(1);
        // unfocus so this works
        min_bound_input.simulate('blur');
        max_bound_input.simulate('blur');
        how_many_input.simulate('blur');
        expect(wrapper.props().updateMinBoundBlur).toHaveBeenCalledTimes(1);
        expect(wrapper.props().updateMaxBoundBlur).toHaveBeenCalledTimes(1);
        expect(wrapper.props().updateHowManyBlur).toHaveBeenCalledTimes(1);

        const decimal_select = wrapper.find({name: 'decimal'});
        decimal_select.simulate('change', {target: {value: '1'}});
        expect(wrapper.props().updateDecimalPlacement).toHaveBeenCalledTimes(1);

        const chn_char_select = wrapper.find({name: 'chinese-character-type'});
        chn_char_select.simulate('change', {target: {value: 'tc'}});
        expect(wrapper.props().updateChnCharType).toHaveBeenCalledTimes(1);
    });
    // click more options 
    test('MenuBoard click MoreOptions button to show or hide MoreOptions', () => {
        const wrapper = Enzyme.mount(
            <MenuBoard
                {...menu_board_props}
            />
        );
        expect(wrapper.find(MoreOptions).exists()).toBe(false);
        
        wrapper.find('#more-options-button').simulate('click')
        expect(wrapper.find(MoreOptions).exists()).toBe(true);
        wrapper.find('#hide-button').simulate('click')
        expect(wrapper.find(MoreOptions).exists()).toBe(false);
    })
});

describe('snapshots', () => {
    test('renders NumberOptions with isMobile false', () => {
        const wrapper = Enzyme.shallow(
            <NumberOptions
                {...number_options_props}
            />
        )
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test('renders NumberOptions with isMobile true', () => {
        number_options_props.isMobile = true;
        const wrapper = Enzyme.shallow(
            <NumberOptions
                {...number_options_props}
            />
        )
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test('renders MoreOptions with isMobile being false ', () => {
        const wrapper = Enzyme.shallow(
            <MoreOptions
                {...more_options_props}        
            />
        )
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('renders MoreOptions with isMobile being true', () => {
        more_options_props.isMobile = true;
        const wrapper = Enzyme.shallow(
            <MoreOptions
                {...more_options_props}        
            />
        )
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test('renders Options with isMobile being false', () => {
        const wrapper = Enzyme.shallow(
            <Options
                {...options_props}
            />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test('renders Options with isMobile being true', () => {
        options_props.isMobile = true
        const wrapper = Enzyme.shallow(
            <Options
                {...options_props}
            />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test('renders MenuBoard with isMobile being false', () => {
        const wrapper = Enzyme.shallow(
            <MenuBoard
                {...menu_board_props}
            />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test('renders MenuBoard with isMobile being true', () => {
        menu_board_props.isMobile = true
        const wrapper = Enzyme.shallow(
            <MenuBoard
                {...menu_board_props}   
            />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
})
