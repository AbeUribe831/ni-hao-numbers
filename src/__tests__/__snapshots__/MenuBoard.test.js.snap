// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots renders MenuBoard with isMobile being false 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <div
    className="welcome-text"
    style={
      Object {
        "alignItems": "flex-end",
        "color": "white",
        "display": "flex",
        "flex": "1",
        "fontSize": "4em",
        "paddingBottom": "0px",
        "textAlign": "center",
      }
    }
  >
    准备好了吗？
  </div>
  <div
    style={
      Object {
        "flex": "3",
      }
    }
  >
    <Options
      chnCharType="sc"
      confirmHowMany={[MockFunction]}
      confirmMaxIsNumber={[MockFunction]}
      confirmMinIsNumber={[MockFunction]}
      decimalPlacement="0"
      howMany="5"
      isMobile={false}
      maxBound="10"
      minBound="0"
      moreOptions={false}
      onClickMoreOptions={[Function]}
      onClickStart={[MockFunction]}
      questions={
        Object {
          "listen": false,
          "readCharacter": true,
          "readNumber": true,
        }
      }
      resetQAndA={[MockFunction]}
      started={false}
      updateAnswers={[MockFunction]}
      updateChnCharType={[MockFunction]}
      updateDecimalPlacement={[MockFunction]}
      updateHowMany={[MockFunction]}
      updateHowManyBlur={[MockFunction]}
      updateMaxBound={[MockFunction]}
      updateMaxBoundBlur={[MockFunction]}
      updateMinBound={[MockFunction]}
      updateMinBoundBlur={[MockFunction]}
      updateQuestions={[MockFunction]}
    />
  </div>
</div>
`;

exports[`snapshots renders MenuBoard with isMobile being true 1`] = `
<div>
  <div
    className="welcome-text"
  >
    准备好了吗？
  </div>
  <div>
    <Options
      chnCharType="sc"
      confirmHowMany={[MockFunction]}
      confirmMaxIsNumber={[MockFunction]}
      confirmMinIsNumber={[MockFunction]}
      decimalPlacement="0"
      howMany="5"
      isMobile={true}
      maxBound="10"
      minBound="0"
      moreOptions={false}
      onClickMoreOptions={[Function]}
      onClickStart={[MockFunction]}
      questions={
        Object {
          "listen": false,
          "readCharacter": true,
          "readNumber": true,
        }
      }
      resetQAndA={[MockFunction]}
      started={false}
      updateAnswers={[MockFunction]}
      updateChnCharType={[MockFunction]}
      updateDecimalPlacement={[MockFunction]}
      updateHowMany={[MockFunction]}
      updateHowManyBlur={[MockFunction]}
      updateMaxBound={[MockFunction]}
      updateMaxBoundBlur={[MockFunction]}
      updateMinBound={[MockFunction]}
      updateMinBoundBlur={[MockFunction]}
      updateQuestions={[MockFunction]}
    />
  </div>
</div>
`;

exports[`snapshots renders MoreOptions with isMobile being false  1`] = `
<div
  className="flex-column"
>
  <p
    className="desktop-sub-titles"
    style={
      Object {
        "alignSelf": "center",
        "color": "rgb(235, 200, 5)",
        "margin": "8px 0px 0px 0px",
      }
    }
  >
    Questions
  </p>
  <div
    className="flex-row margin-top"
  >
    <button
      className="standard-button"
      id="read-character"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#cd071e",
          "color": "white",
          "transitionDuration": "0.1s",
        }
      }
    >
      read character
    </button>
    <div
      style={
        Object {
          "marginLeft": "0.5em",
        }
      }
    />
    <button
      className="standard-button"
      id="read-number"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#cd071e",
          "color": "white",
          "transitionDuration": "0.1s",
        }
      }
    >
      read number
    </button>
    <div
      style={
        Object {
          "marginLeft": "0.5em",
        }
      }
    />
    <button
      className="standard-button"
      id="listen"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "color": "#cd071e",
          "transitionDuration": "0.1s",
        }
      }
    >
      listen
    </button>
  </div>
  <div
    className="margin-top"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <button
      className="underline-button"
      id="hide-button"
      onClick={[MockFunction]}
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      hide
    </button>
  </div>
</div>
`;

exports[`snapshots renders MoreOptions with isMobile being true 1`] = `
<div
  className="flex-column"
>
  <p
    className=""
    style={
      Object {
        "alignSelf": "center",
        "color": "rgb(235, 200, 5)",
        "margin": "8px 0px 0px 0px",
      }
    }
  >
    Questions
  </p>
  <div
    className="flex-row margin-top"
  >
    <button
      className="mobile-standard-button"
      id="read-character"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#cd071e",
          "color": "white",
          "transitionDuration": "0.1s",
        }
      }
    >
      read character
    </button>
    <div
      style={
        Object {
          "marginLeft": "0.5em",
        }
      }
    />
    <button
      className="mobile-standard-button"
      id="read-number"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#cd071e",
          "color": "white",
          "transitionDuration": "0.1s",
        }
      }
    >
      read number
    </button>
    <div
      style={
        Object {
          "marginLeft": "0.5em",
        }
      }
    />
    <button
      className="mobile-standard-button"
      id="listen"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "color": "#cd071e",
          "transitionDuration": "0.1s",
        }
      }
    >
      listen
    </button>
  </div>
  <div
    className="margin-top"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <button
      className="underline-button"
      id="hide-button"
      onClick={[MockFunction]}
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      hide
    </button>
  </div>
</div>
`;

exports[`snapshots renders NumberOptions with isMobile false 1`] = `
<div
  className="flex-column"
>
  <div
    className="flex-row margin-top"
  >
    <div
      className="flex-column"
    >
      <p
        className="desktop-sub-titles"
      >
        From
      </p>
      <input
        className="desktop-input"
        id="min-bound"
        name="minBound"
        onBlur={[MockFunction]}
        onChange={[MockFunction]}
        onKeyDown={[MockFunction]}
        value="0"
      />
    </div>
    <div
      style={
        Object {
          "marginLeft": "1em",
        }
      }
    />
    <div
      className="flex-column"
    >
      <p
        className="desktop-sub-titles"
      >
        To
      </p>
      <input
        className="desktop-input"
        id="max-bound"
        name="maxBound"
        onBlur={[MockFunction]}
        onChange={[MockFunction]}
        onKeyDown={[MockFunction]}
        value="10"
      />
    </div>
    <div
      style={
        Object {
          "marginLeft": "1em",
        }
      }
    />
    <div
      className="flex-column"
    >
      <p
        className="desktop-sub-titles"
      >
        How Many
      </p>
      <input
        className="desktop-input"
        id="how-many"
        name="howMany"
        onBlur={[MockFunction]}
        onChange={[MockFunction]}
        onKeyDown={[MockFunction]}
        value="5"
      />
    </div>
  </div>
  <div
    className="flex-row margin-top"
  >
    <div
      className="flex-column"
    >
      <p
        className="desktop-sub-titles"
      >
        Decimal
      </p>
      <select
        className="desktop-input"
        id="decimal"
        name="decimal"
        onChange={[MockFunction]}
        value="0"
      >
        <option
          value="0"
        >
          0
        </option>
        <option
          value="1"
        >
          1
        </option>
        <option
          value="2"
        >
          2
        </option>
      </select>
    </div>
    <div
      style={
        Object {
          "marginLeft": "1em",
        }
      }
    />
    <div
      className="flex-column"
    >
      <p
        className="desktop-sub-titles"
      >
        Characters
      </p>
      <select
        className="desktop-input"
        id="chinese-character-type"
        name="chinese-character-type"
        onChange={[MockFunction]}
        value="sc"
      >
        <option
          value="sc"
        >
          简体
        </option>
        <option
          value="tc"
        >
          繁體
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`snapshots renders NumberOptions with isMobile true 1`] = `
<div
  className="flex-column"
>
  <div
    className="flex-row margin-top"
  >
    <div
      className="flex-column"
    >
      <p
        className="sub-titles"
      >
        From
      </p>
      <input
        className=""
        id="min-bound"
        name="minBound"
        onBlur={[MockFunction]}
        onChange={[MockFunction]}
        onKeyDown={[MockFunction]}
        value="0"
      />
    </div>
    <div
      style={
        Object {
          "marginLeft": "1em",
        }
      }
    />
    <div
      className="flex-column"
    >
      <p
        className="sub-titles"
      >
        To
      </p>
      <input
        className=""
        id="max-bound"
        name="maxBound"
        onBlur={[MockFunction]}
        onChange={[MockFunction]}
        onKeyDown={[MockFunction]}
        value="10"
      />
    </div>
    <div
      style={
        Object {
          "marginLeft": "1em",
        }
      }
    />
    <div
      className="flex-column"
    >
      <p
        className="sub-titles"
      >
        How Many
      </p>
      <input
        className=""
        id="how-many"
        name="howMany"
        onBlur={[MockFunction]}
        onChange={[MockFunction]}
        onKeyDown={[MockFunction]}
        value="5"
      />
    </div>
  </div>
  <div
    className="flex-row margin-top"
  >
    <div
      className="flex-column"
    >
      <p
        className="sub-titles"
      >
        Decimal
      </p>
      <select
        className=""
        id="decimal"
        name="decimal"
        onChange={[MockFunction]}
        value="0"
      >
        <option
          value="0"
        >
          0
        </option>
        <option
          value="1"
        >
          1
        </option>
        <option
          value="2"
        >
          2
        </option>
      </select>
    </div>
    <div
      style={
        Object {
          "marginLeft": "1em",
        }
      }
    />
    <div
      className="flex-column"
    >
      <p
        className="sub-titles"
      >
        Characters
      </p>
      <select
        className=""
        id="chinese-character-type"
        name="chinese-character-type"
        onChange={[MockFunction]}
        value="sc"
      >
        <option
          value="sc"
        >
          简体
        </option>
        <option
          value="tc"
        >
          繁體
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`snapshots renders Options with isMobile being false 1`] = `
<div
  className="flex-column"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <NumberOptions
    chnCharType="sc"
    confirmHowMany={[MockFunction]}
    confirmMaxIsNumber={[MockFunction]}
    confirmMinIsNumber={[MockFunction]}
    decimalPlacement="0"
    howMany="6"
    isMobile={false}
    maxBound="10"
    minBound="0"
    updateChnCharType={[MockFunction]}
    updateDecimalPlacement={[MockFunction]}
    updateHowMany={[MockFunction]}
    updateMaxBound={[MockFunction]}
    updateMinBound={[MockFunction]}
  />
  <div
    className="margin-top"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <button
      className="underline-button"
      id="more-options-button"
      onClick={[MockFunction]}
    >
      more options
    </button>
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <button
      className="standard-button"
      id="start-button"
      onClick={[MockFunction]}
      style={
        Object {
          "backgroundColor": "#cd071e",
          "color": "white",
          "cursor": "pointer",
          "type": "button",
        }
      }
    >
      start
    </button>
  </div>
</div>
`;

exports[`snapshots renders Options with isMobile being true 1`] = `
<div
  className="flex-column"
  style={Object {}}
>
  <NumberOptions
    chnCharType="sc"
    confirmHowMany={[MockFunction]}
    confirmMaxIsNumber={[MockFunction]}
    confirmMinIsNumber={[MockFunction]}
    decimalPlacement="0"
    howMany="6"
    isMobile={true}
    maxBound="10"
    minBound="0"
    updateChnCharType={[MockFunction]}
    updateDecimalPlacement={[MockFunction]}
    updateHowMany={[MockFunction]}
    updateMaxBound={[MockFunction]}
    updateMinBound={[MockFunction]}
  />
  <div
    className="margin-top"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <button
      className="underline-button"
      id="more-options-button"
      onClick={[MockFunction]}
    >
      more options
    </button>
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <button
      className="mobile-standard-button"
      id="start-button"
      onClick={[MockFunction]}
      style={
        Object {
          "backgroundColor": "#cd071e",
          "color": "white",
          "cursor": "pointer",
          "type": "button",
        }
      }
    >
      start
    </button>
  </div>
</div>
`;
